### ISSUE TITLE
dialog: defaults to opacity:0 and hides

### ISSUE THREAD

**Issue by kschwidder** [angular/material#4862]
When configuring a dialog as below it happens that the opacity:0 is set.
The dialog is shown behind the scroll-mask and therefore not visible.
Sometimes the opacity is set to 1 but when using confirm it is set to 0.

```js
function dialog(event) {
    var confirm = $mdDialog.confirm()
        .title($translate.instant('DIALOG_TITLE_SKILL_DELETE'))
        .content('DELETE')
        .ariaLabel($translate.instant('STANDARD_BTN_DELETE'))
        .ok($translate.instant('STANDARD_BTN_DELETE'))
        .cancel($translate.instant('STANDARD_BTN_CANCEL'))
        .targetEvent(event);

    $mdDialog.show(confirm).then(function() {
        deleteSkill();
    }, function() {
    });
}

element.bind('click', function (event) {
    if (hasAccessRight) {
        dialog(event);
    }
});
```

Angular 1.4.6, Material 0.11.1

**Comment by kschwidder** [comment]
When looking at the material.css the md-dialog defines the opacity:0 as default.
My workaround is to set the opacity:1. Now all dialogs are shown properly so far.
Please investigate and fix urgently as this is major show stopper.

**Comment by nkoterba** [comment]
I think I can further elaborate on this issue.

It appears that in the latest version of MD, the opacity of the dialog by default is 0.

On an animation transition, the opacity becomes 1 and the dialog appears.

Here is the MD css:
```css
md-dialog {
  opacity: 0;
  min-width: 240px;
  max-width: 80%;
  max-height: 80%;
  position: relative;
  overflow: auto;
  box-shadow: 0px 7px 8px -4px rgba(0, 0, 0, 0.2), 0px 13px 19px 2px rgba(0, 0, 0, 0.14), 0px 5px 24px 4px rgba(0, 0, 0, 0.12);
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column; }

md-dialog.md-transition-in {
  opacity: 1;
  transition: all 0.4s cubic-bezier(0.25, 0.8, 0.25, 1);
  -webkit-transform: translate3d(0, 0, 0) scale(1);
  transform: translate3d(0, 0, 0) scale(1); }

md-dialog.md-transition-out {
  opacity: 0;
  transition: all 0.4s cubic-bezier(0.25, 0.8, 0.25, 1);
  -webkit-transform: translate3d(0, 100%, 0) scale(0.2);
  transform: translate3d(0, 100%, 0) scale(0.2); }
```

The issue that arises here is what if someone disables Angular's animations?  E.g. in development, we have an option in our `gulp` task that sets a property to enable/disable animations.  Then in our angular application we do:
```javascript
		.run(['$animate', '$log', 'useAnimations', function ($animate, $log, useAnimations) {
			$log.info('Animations enabled: ', useAnimations);

			$animate.enabled(useAnimations);
		}])
```

Since our animations are disabled, the dialog is never "animated" to an opacity of 1.

**Comment by kschwidder** [comment]
+1. 
This is what happens in our case. we have disabled animation.

**Comment by denistrofimov** [comment]
Fixed in 0.11.2?

**Comment by apat183** [comment]
+1 this is happen for me. It's only happening on confirm dialog and not custom.
